### PR TITLE
Update framework

### PR DESCRIPTION
**Pull request by peppy**
Includes required changes for new URL safeties.

- Includes and closes #21652 
- Includes and closes #17462

### CODE REVIEW

**Comment by peppy** [comment]
This one's going to require the fixes to be applied for android (and potentially iOS) as a result of framework-side changes.

**Comment by peppy** [comment]
@frenzibyte i'll leave the mobile projects to you to figure out. looks like the `csproj` needs updating at very least.

**Comment by frenzibyte** [comment]
Should be fixed now, looks to be happening due to GH actions not yet using Xcode 14.1, so I've had to manually pin it to do so. See https://github.com/actions/runner-images/issues/6771 for more information.

**Comment by peppy** [comment]
There's a theory that this may be ready to go now, if CI passes. @smoogipoo would appreciate a one-over (feel free to apply any changes you deem necessary directly).

**Comment by smoogipoo** [comment]
Wanna wait for new o!f package with https://github.com/ppy/osu-framework/pull/5592 before merging this one?

**Comment by peppy** [comment]
Well this one works fine as-is, since none of the veldrid stuff is in.

So I figure it's best to review / merge this first then we'll do a second bump. But if you'd rather do it all at once I'm fine with that too.

**Review by smoogipoo — APPROVED** [review]

File: global.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "sdk": {
+    "version": "6.0.300",
+    "rollForward": "major"
+  }
```
Comment (Morilli) [inline global.json:5]: This was removed in #17462 but still exists here, was that intentional?

Comment (peppy) [reply global.json:5]: Was an oversight, I'll pull it into a follow-up branch I have.

**Pull request merged**